### ISSUE TITLE
missing varfont api: listInstances

### ISSUE THREAD

**Issue by typemytype** [typemytype/drawbot#256]
requested here https://forum.drawbot.com/topic/121/listinstances-font

**Comment by typemytype** [comment]
this should do it

```python
import CoreText
import AppKit

from fontTools.ttLib import TTFont

from drawBot.context.tools import variation


def getIntancesForFontName(fontName):
    instances = {}
    font = CoreText.CTFontCreateWithName(fontName, 12, None)
    if font is None:
        return instances
    cgFont = CoreText.CGFontCreateWithFontName(fontName)
    fontDescriptor = font.fontDescriptor()
    url = CoreText.CTFontDescriptorCopyAttribute(fontDescriptor, CoreText.kCTFontURLAttribute)
    if url is None:
        return instances

    variationAxesDescriptions = CoreText.CTFontCopyVariationAxes(font)
    tagNameMap = {}
    for variationAxesDescription in variationAxesDescriptions:
        tag = variation.convertIntToVariationTag(variationAxesDescription[CoreText.kCTFontVariationAxisIdentifierKey])
        name = variationAxesDescription[CoreText.kCTFontVariationAxisNameKey]
        tagNameMap[tag] = name

    ft = TTFont(url.path())
    if "fvar" in ft:
        fvar = ft["fvar"]

        for instance in fvar.instances:
            fontVariations = dict()
            for axis, value in instance.coordinates.items():
                fontVariations[tagNameMap[axis]] = value

            varFont = CoreText.CGFontCreateCopyWithVariations(cgFont, fontVariations)   
            postScriptName = CoreText.CGFontCopyPostScriptName(varFont)
            instances[postScriptName] = instance.coordinates
            
    ft.close()
    return instances

r = getIntancesForFontName("Skia")
print(r)
```

**Comment by justvanrossum** [comment]
This has been implemented in #259.
